lib/component.js

```diff
@@ -616,12 +623,19 @@ class Component extends WebComponent {
       const attrType = attrSchema.type;
       let attrValue = null;
 
+      if (attrSchema.deprecatedMsg) {
+        console.warn(`${this}: attr '${attr}' is deprecated. ${attrSchema.deprecatedMsg}`);
+      }
+
```

Comment:
Moved the deprecation message up due to the new early return for missing required attributes.